### ISSUE TITLE
Add a way to know which slot a dynamic impulse came from

### ISSUE THREAD

**Issue by Nytra** [Yellow-Dog-Man/Resonite-Issues#278]
### Is your feature request related to a problem? Please describe.

It would be useful to know which slot a dynamic impulse came from because then you could make functions which return data to the calling slot or perform some operations on it. I know you can use DynamicImpulseTriggerWithObject<Slot> to do this but then you lose out on being able to send other input data.

### Describe the solution you'd like

Add a way to know which slot a dynamic impulse came from. Maybe on the dynamic impulse receiver there could be a output for the calling slot. This would be in addition to any other data, so if you had a DynamicImpulseReceiverWithValue<float> you would be able to get the float output and also the calling slot output.

### Describe alternatives you've considered

Using DynamicImpulseTriggerWithObject<Slot> to send the calling slot but then you can't send other input data at the same time unless you use other methods.

**Comment by moonheart08** [comment]
Wouldn't the best alternative be the introduction of DynamicImpulseTrigger<T1, T2>? Which'd just let you send both the slot and some data.

**Comment by Nytra** [comment]
Yes being able to trigger dynamic impulses with multiple arguments of any type would be great. However I feel as though the calling slot would be a common thing that people would want to use so maybe better to just make that a default on every dynamic impulse.

**Comment by JackTheFoxOtter** [comment]
This is a really cool idea. I like that. The issue is just that the calling slot would be the node itself, so sending an impulse directly back to it wouldn't be very desirable in pretty much all cases. I think a more generic solution would be specifying what slot to send through as well, but that would require more parameters, and I assume that would require collection support.

**Comment by Nytra** [comment]
Even if it just gave you the node slot it could still be useful, because if that slot is inside a dynamic variable space you can use Write Dynamic Variable on it and it would correctly write data to the space. You could also do things like verify that the slot calling the impulse is actually inside of a certain hierarchy, so that anything outside that hierarchy trying to call the impulse would just get rejected. So it could add a bit of security to dynamic impulses as well.

**Comment by Frooxius** [comment]
You should probably just send the slot as value.

With some future additions, like nested nodes or static ProtoFlux assets, impulses might not come from a slot at all, so this would set an odd expectation/predecent.

**Comment by Nytra** [comment]
In that case, would it be possible to allow sending multiple values/objects with a dynamic impulse? I think this might require collections though, but I'm sure, since we do have SyncList/SyncRefList already which could be used. Also there are multiple nodes which have buttons to change the number of inputs and outputs.

**Comment by Frooxius** [comment]
Yes, with collections you'd be able to do this. You can just pass multiple things in a dictionary for example.
